src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -497,7 +498,8 @@ TypeDropIndexConstraint(const AlterTableCmd *command,
 		{
 			/* we found the constraint, now check if it is for an index */
 			if (constraintForm->contype == CONSTRAINT_PRIMARY ||
-				constraintForm->contype == CONSTRAINT_UNIQUE)
+				constraintForm->contype == CONSTRAINT_UNIQUE ||
+				constraintForm->contype == CONSTRAINT_EXCLUSION)
```

Comment:
@aamederen The data structures, and even the enums, differ. One is from the parse tree, with a `ConstrType` enum. The other is after a catalog lookup, checking the constraint type as a char. May we skip this refactoring in the PR?